### PR TITLE
save and load optimizer stats along with the model

### PR DESCRIPTION
**Pull request by pandeydeep9**

### CODE REVIEW

File: experiment_scripts/omniglot_maml-omniglot_1_8_0.1_64_5_0_few_shot.sh [hunk 1]
```diff
@@ -1,11 +1,11 @@
 #!/bin/sh
 
-export GPU_ID=$1
+export GPU_ID=1
```
Comment (AntreasAntoniou) [inline experiment_scripts/omniglot_maml-omniglot_1_8_0.1_64_5_0_few_shot.sh:3]: This modification is wrong. What you want is $1, which passes the second argument entered into the terminal after the script name and sets it as the GPU idx that the program should be using.

Comment (AntreasAntoniou) [reply experiment_scripts/omniglot_maml-omniglot_1_8_0.1_64_5_0_few_shot.sh:3]: If you agree to this change, please modify the file and I'll accept your PR.

File: experiment_config/omniglot_maml-omniglot_1_8_0.1_64_5_0.json [hunk 2]
```diff
@@ -37,6 +37,7 @@
   "weight_decay": 0.0,
   "dropout_rate_value":0.0,
   "min_learning_rate":0.00001,
+  "task_learning_rate":0.1,
```
Comment (AntreasAntoniou) [inline experiment_config/omniglot_maml-omniglot_1_8_0.1_64_5_0.json:40]: Was this missing in the original file? Does it cause a runtime error?

Comment (pandeydeep9) [reply experiment_config/omniglot_maml-omniglot_1_8_0.1_64_5_0.json:40]: Yes this was missing in the original json file. It does not cause any error. But, in your utils/parser_utils.py this is default to 0.1 (this might be useful I think)

File: experiment_scripts/omniglot_maml-omniglot_1_8_0.1_64_5_0_few_shot.sh (hunk 1 above)
Comment (pandeydeep9) [reply experiment_scripts/omniglot_maml-omniglot_1_8_0.1_64_5_0_few_shot.sh]: Sure, I have modified it. Yes, it makes sense to keep it to $1. Thanks

File: experiment_config/omniglot_maml-omniglot_1_8_0.1_64_5_0.json (hunk 2 above)
Comment (pandeydeep9) [reply experiment_config/omniglot_maml-omniglot_1_8_0.1_64_5_0.json:40]: Basically, for MAML experiments with miniImagenet dataset, I wanted to set the inner loop learning rate to 0.01/0.001 and felt adding this to json will help. I think this should be added to all other .json files as well.

**Review by AntreasAntoniou — APPROVED** [review]
